### ISSUE TITLE
itsycal aborting with segv, warning message about dispatch handler

### ISSUE THREAD

**Issue by resuna** [sfsam/Itsycal#124]
I noticed itsyCal was no longer in my menu bar. I looked for log info and got this:

Mar 13 07:24:40 fa-peterdasilva com.apple.xpc.launchd[1] (com.mowglii.ItsycalApp.13136[86948]): Service exited due to SIGSEGV | sent by exc handler[86948]

When I restarted ItsyCal I got this:

Mar 13 15:09:30 fa-peterdasilva Itsycal[9155]: DEPRECATED USE in libdispatch client: dispatch source activated with no event handler set; set a breakpoint on _dispatch_bug_deprecated to debug

And when I looked for crash reports:

```
Process:               Itsycal [902]
Path:                  /Applications/Itsycal.app/Contents/MacOS/Itsycal
Identifier:            com.mowglii.ItsycalApp
Responsible:           Itsycal [902]
Crashed Thread:        3  Dispatch queue: com.mowglii.Itsycal.eventQueue
    __TEXT                 0000000103785000-00000001037ba000 [  212K] r-x/rwx SM=COW  /Applications/Itsycal.app/Contents/MacOS/Itsycal
21  com.mowglii.ItsycalApp        	0x000000010378b383 -[EventCenter fetchSourcesAndCalendars] + 71
22  com.mowglii.ItsycalApp        	0x000000010378d67a -[EventCenter refetchAll] + 68
Thread 3 Crashed:: Dispatch queue: com.mowglii.Itsycal.eventQueue
2   com.mowglii.ItsycalApp        	0x000000010378cfd0 -[EventCenter fetchEventsWithStartDate:endDate:] + 3005
3   com.mowglii.ItsycalApp        	0x000000010378c27e __26-[EventCenter fetchEvents]_block_invoke + 295
       0x103785000 -        0x1037b9ffb +com.mowglii.ItsycalApp (0.11.13 - 1079) <B53806D7-6712-3FF7-A546-A816D872E55C> /Applications/Itsycal.app/Contents/MacOS/Itsycal
       0x1037f5000 -        0x103802fff +com.github.shpakovski.MASShortcut (2.3.6 - 2.3.6) <657566BC-0238-3AFE-8040-2D82E7312311> /Applications/Itsycal.app/Contents/Frameworks/MASShortcut.framework/Versions/A/MASShortcut
       0x103814000 -        0x103817ffb +com.mowglii.MoLoginItem (1.0 - 1) <73054342-9ADF-391B-82B3-1182569CC2CC> /Applications/Itsycal.app/Contents/Frameworks/MoLoginItem.framework/Versions/A/MoLoginItem
       0x103825000 -        0x103867fff +org.sparkle-project.Sparkle (1.18.0 22-g7fa5b1d6 - 1.18.0) <16276C0E-DAC7-3763-A77B-31D75DB868B2> /Applications/Itsycal.app/Contents/Frameworks/Sparkle.framework/Versions/A/Sparkle
```

Which sounds suspiciously like it needs some updating to not use the deprecated API.

**Comment by sfsam** [comment]
@resuna Could you send me the crash reports to mowgliiapps@gmail.com? They should be in ~/Library/Logs/DiagnosticReports/.

I got reports of a crash which may be the one you are seeing and I have a test version for that crash in case you are able to try it out. It might be the same one you got.

http://s3.amazonaws.com/itsycal/Itsycal-0.11.13-crash-test-1.zip

If you are still using Itsycal, you'll have to quit it first and replace the version you have in the /Applications folder with this new one.

If you do try this out, could you let me know how it goes?

**Comment by resuna** [comment]
I sent you the crash reports.

I will let you know how it goes, the crash is not super replicable... it can go days without crashing, or it can die multiple times a day. I have no problem at home under Sierra.

**Comment by sfsam** [comment]
@resuna Any crashes with the test version so far?

**Comment by resuna** [comment]
Nope!

**Comment by sfsam** [comment]
Thanks. I'll close this as I believe it's fixed in v0.11.14.
